dynamic_programming/LCS.c

```diff
@@ -0,0 +1,212 @@
+/**
+ * @file
+ * @author [Santosh Jonnakuti](https://github.com/Santoshjonnakuti)
+ * @brief [Longest Common Subsequence Algorithm](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem)
+ * implementation in C to find the LCS between 2 strings.
+ * @details LCS algorithm uses dynamic approach to find all possible LCS between 2 strings. 
+ * Dynamic Programming follow a method so that even sub-problems will also have optimal solution.
+ *
+ * LCS() will find the length of the LCS of the following two strings 'GAC' and 'AGCAT'.
+ * LCS function will return the follwing 2D-Matrix
+ *
+ *	  0 0 0 0 0 0
+ *      0 0 1 1 1 1
+ *      0 1 1 1 2 2
+ *      0 1 1 2 2 2
+ *
+ * So Length of LCS will be 2.
+ *
+ * allPossibleLSC() will print all possible LCS between those strings.
+ *
+ * AC
+ * GC
+ * GA
+ *
+ * So There are 3 Longest common subsequences in 'GAC' and 'AGCAT'
+ */
+
+
+#include<stdio.h>     /// for I/O operations
+#include<stdlib.h>   ///  for Dynamic memory allocation
+#include<string.h>  ///   for puts(), gets() and strlen()
+
+#define MAX 100      // Declaration of a constant value
+
+int lengthOfLCS;
+
+// Functions Declaration
+int max(int a, int b);
+int** LCS(char *string1, int m, char *string2, int n);
+char* allPossibleLCS(int **c, char *a, int index, int i, int j, char *string1, char *string2);
+void print(char *a, int index);
+
+/**
+ * @brief prints all possible Longest common subsequences between string1 and string2 using recursion
+ * @param c 2D-Array of Optimal solutions of all sub-problems
+ * @param a Longest common subsequence 
+ * @param index  length of a
+ * @param i length of string 1
+ * @param j length of string 2
+ * @param string1 String 1
+ * @param string2 String 2
+ * @prints one of te Longest Common Subsequence
```

Comment:
What does this function return? Every function returns something unless they are `void`. Please document it here.